gnomad/utils/sparse_mt.py

```diff
@@ -1050,20 +1050,28 @@ def join_with_ref(mt: hl.MatrixTable) -> hl.MatrixTable:
     mean_expr = hl.agg.mean(hl.or_else(mt.DP, 0))
 
     # Annotate rows now
-    return mt.select_rows(
-        mean=hl.if_else(hl.is_nan(mean_expr), 0, mean_expr),
-        median_approx=hl.or_else(hl.agg.approx_median(hl.or_else(mt.DP, 0)), 0),
-        total_DP=hl.agg.sum(mt.DP),
-        **{
-            f"over_{x}": count_array_expr[i] / n_samples
-            for i, x in zip(
-                range(
-                    len(coverage_over_x_bins) - 1, -1, -1
-                ),  # Reverse the bin index as count_array_expr has the reverse order
-                coverage_over_x_bins,
-            )
-        },
-    ).rows()
+    ht = (
+        mt.select_rows(
+            mean=hl.if_else(hl.is_nan(mean_expr), 0, mean_expr),
+            median_approx=hl.or_else(hl.agg.approx_median(hl.or_else(mt.DP, 0)), 0),
+            total_DP=hl.agg.sum(mt.DP),
+            **{
+                f"over_{x}": count_array_expr[i] / n_samples
+                for i, x in zip(
+                    range(
+                        len(coverage_over_x_bins) - 1, -1, -1
+                    ),  # Reverse the bin index as count_array_expr has the reverse order
+                    coverage_over_x_bins,
+                )
+            },
+        )
+        .rows()
+        .key_by("locus")
+        .select_globals()
+        .drop("alleles")
+    )
+
```

Comment:
I think it should be keyed by `*row_key_fields` and not drop alleles within this function, the user should do that if wanted

Reply:
i edited this to rekey by row_key_fields and drop any previous keys, not sure if that's what's wanted, can also drop alleles after this line in gnomad_qc (https://github.com/broadinstitute/gnomad_qc/blob/dbd66b79bb02a4a3bf73c4541de14a0eccf8e98c/gnomad_qc/v4/annotations/compute_coverage.py#L122C14-L122C14)

Reply:
Ah, yes, I think this is good. I was thinking to drop in gnomad_qc, but I think this makes more sense